app/routes.py

```diff
@@ -1,2 +1,344 @@
+
+        if task_query:
+            if task_query == "asc":
+                tasks = Task.query.order_by(Task.title.asc())
+            elif task_query == "desc":
+                tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append(
+                {
+                    "id" : task.task_id,
+                    "title" : task.title,
+                    "description" : task.description,
+                    "is_complete" : task.is_complete
+                }
+            )
+        
+        return jsonify(tasks_response), 200
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+
+        #replace with try/except
+        #this is my guard clause
+        if (("title" not in request_body.keys()) or 
+        ("description" not in request_body.keys()) or
+        ("completed_at" not in request_body.keys())):
+            return jsonify(
+                {"details": "Invalid data"}
+            ), 400
+
+        if not request_body["completed_at"]:
+            new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=request_body["completed_at"],
+                            )
+            db.session.add(new_task)
+            db.session.commit()
+
+
+            return jsonify({"task":
+                {
+                "id" : new_task.task_id,
+                "title" : new_task.title,
+                "description" : new_task.description,
+                "is_complete" : new_task.is_complete
+                    }
+                }), 201
+
+        elif request_body["completed_at"]:
+            new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=request_body["completed_at"],
+                            is_complete=True
+                            )
+            db.session.add(new_task)
+            db.session.commit()
+
+
+            return jsonify({"task":
+                {
+                "id" : new_task.task_id,
+                "title" : new_task.title,
+                "description" : new_task.description,
+                "is_complete" : new_task.is_complete
+                    }
+                }), 201
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+
+    if request.method == "GET":
+        
+        return_dict = {"task":
+            {
+            "id" : task.task_id,
+            "title" : task.title,
+            "description" : task.description,
+            "is_complete" : task.is_complete
+                }
+            }
+
+        if task.goal_id:
+            return_dict["task"]["goal_id"] = task.goal_id
```

Comment:
Should the route be the one that decides which model rendering function to use by inspecting the goal relationship within the task itself? Or should the task simply be asked to convert itself to a dictionary, and then it decides how to build the dictionary given its own internal state? It's not clear either way, but it's definitely worth thinking about!